Assets/Scripts/StudioX/UploadManager.cs

```diff
@@ -0,0 +1,46 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using NativeFilePickerNamespace;
+using AsImpL;
+public class UploadManager : MonoBehaviour
+{
+    private string path;
+    private GameObject prefab;
+    private string[] fileTypes;
+    private ObjectImporter obj;
+    private PathSettings pathSettings;
+    private ImportOptions importOptions = new ImportOptions();
+    void Start()
+    {
+        fileTypes = allowedFileTypes();
+        obj = gameObject.GetComponent<ObjectImporter>();
+        pathSettings = gameObject.GetComponent<PathSettings>();
+    }
+    public void LoadLocalModel()
+    {
+        Instantiate(prefab);
+
+
+    }
+    public void SelectModelFile()
+    {
+        if (NativeFilePicker.IsFilePickerBusy())
+            return;
+
+        NativeFilePicker.Permission permission = NativeFilePicker.PickFile((dir) =>
+       {
+           if (dir == null)
+               Debug.Log("Operation cancelled");
```

Comment:
This code could probably be short-circuited with a return statement directly after Debug.Log("Operation cancelled") if someone cancels, otherwise an empty string will be concatenated to pathSettings.RootPath and depending on the ImportModelAsync method may throw an uncaught exception, although I haven't tried it out locally so I'm not sure how it handles OBJ files that aren't found.

Reply:
Makes sense. I'm deving the android part for now and for now I'm just going to validate the string to make sure it ends with obj else it gives a log error. Thanks!